nixio/tag.py

```diff
@@ -113,14 +119,44 @@ def create_feature(self, data, link_type):
         feat = Feature.create_new(self.file, self, features, data, link_type)
         return feat
 
+    def _calc_data_slices(self, data, position, extent, stop_rule):
+        refslice = list()
+        if not self.units:
+            units = [None]*len(data.dimensions)
+        else:
+            units = self.units
+
+        for idx, dim in enumerate(data.dimensions):
+            if idx < len(position):
+                pos = position[idx]
+                start = self._pos_to_idx(pos, units[idx], dim, SliceMode.Inclusive)
+            else:
+                # Tag doesn't specify (pos, ext) for all dimensions: will return entire remaining dimensions (0:len)
+                start = 0
+            if extent is None or len(extent) == 0:
+                # no extents: return one element
+                stop = start + 1
+            elif idx < len(extent):
+                ext = extent[idx]
+                stop = self._pos_to_idx(pos+ext, units[idx], dim, stop_rule) + 1
```

Comment:
why the +1?

Reply:
The `stop` element of a slice isn't inclusive ever.  `Slice(0, 30)` is equivalent to `[0:30]` which returns 30 elements, the last one being index 29.  `_pos_to_idx()` on the `pos+ext` value returns the `last_index`, the index of the last value that should be included in the return slice, so the slice stop value should be `last_index + 1`.

Reply:
got it, thanks